// Copyright 2023 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// Keep in sync with corresponding messages in :
// http://google3/privacy/net/common/proto/get_proxy_config.proto

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package ip_protection;

message GetProxyConfigRequest {
  // Descriptor of which service is calling this endpoint, eg. chromeipblinding.
  string service_type = 1;
}

message GetProxyConfigResponse {
  reserved 2;

  // Chain of proxies.
  message ProxyChain {
    string proxy_a = 1;
    string proxy_b = 2;
  }

  // Hostnames for the first proxy hop. Ordered based on preference for use.
  repeated string first_hop_hostnames = 1 [deprecated = true];
  // Proxy chain each request should be routed through.
  repeated ProxyChain proxy_chain = 3;
}
